site/app/controllers/DockerInterfaceController.php

```diff
@@ -153,6 +155,39 @@ public function addImage() {
         }
     }
 
+    /**
+     * @Route("/admin/remove_image", methods={"POST"})
+     * @return JsonResponse
+     */
+    public function removeImage() {
+        $user = $this->core->getUser();
+        $user_id = $this->core->getUser()->getId();
+        $jsonFilePath = FileUtils::joinPaths(
+            $this->core->getConfig()->getSubmittyInstallPath(),
+            "config",
+            "autograding_containers.json"
+        );
+        $json = json_decode(file_get_contents($jsonFilePath), true);
+        $key = array_search($_POST['image'], $json[$_POST['capability']]);
```

Comment:
The `$_POST['capability']` and `$_POST['capability']` should be tested if they're actually set and is a string. Otherwise a user could use a client like postman and spoof the POST body with null / bad data.

We could also do some validation by checking if the image is a valid name, see some restrictions of what a valid name is here https://docs.docker.com/engine/reference/commandline/tag/#description

but this isn't super necessary, checking its some sort of string I think is good for this first implementation